src/applications/apps.js

```diff
@@ -31,24 +30,20 @@ export function registerApplication(appName, arg1, arg2, customProps = {}) {
     throw new Error(`The first argument must be a non-empty string 'appName'`);
   if (getAppNames().indexOf(appName) !== -1)
     throw new Error(`There is already an app declared with name ${appName}`);
-  if (typeof customProps !== 'object')
+  if (typeof customProps !== 'object' || Array.isArray(customProps))
```

Comment:
This was allowing arrays to pass validation. Now it will correctly fail an array as the customProps.